util/fmtutil/format.go

```diff
@@ -133,25 +133,23 @@ func makeTimeseries(wr *prompb.WriteRequest, labels map[string]string, m *dto.Me
 		toTimeseries(wr, labels, timestamp, float64(m.GetSummary().GetSampleCount()))
 
 	case m.Histogram != nil:
-		metricName := labels[model.MetricNameLabel]
-		// Preserve metric name order with first bucket suffix timeseries then sum suffix timeserie and finally count suffix timeserie
-		// Add Histogram bucket timeseries
-		bucketLabels := make(map[string]string, len(labels)+1)
-		for key, value := range labels {
-			bucketLabels[key] = value
+		h := m.GetHistogram()
+
+		if isNativeHistogram(h) {
+			addNativeHistogramToWriteRequest(wr, h, labels, timestamp)
 		}
-		for _, b := range m.GetHistogram().Bucket {
-			bucketLabels[model.MetricNameLabel] = metricName + bucketStr
-			bucketLabels[model.BucketLabel] = fmt.Sprint(b.GetUpperBound())
-			toTimeseries(wr, bucketLabels, timestamp, float64(b.GetCumulativeCount()))
+
+		// While a quantile-less summary is preferred over a bucket-less histogram, the
+		// latter could happen in principle. The other two properties of a classic histogram
+		// (sum and count) are also set for native histograms, so the only signal we have to
+		// decide whether the classic histogram should be included in the request is having
+		// a list of buckets.
+		//
+		// Note that this comes _after_ native histogams because calling
+		// addClassicHistogramToWriteRequest modifies the labels map.
+		if len(h.Bucket) > 0 {
```

Comment:
I think this should be `if len(h.Bucket) > 0 || !isNativeHistogram(h) {`

Rationale: If there are classic buckets, then yes, you want this to be represented as a classic histogram, maybe even in parallel with the native histogram.

However, even if there are no buckets, this might be the edge case explained in your doc comment above. We can only say for sure if it is _not_ a native histogram. So in that case we should actually create that (arguably incomplete) classic histogram.